src/test/java/SixPMComTestSuite/SixPMFeaturesTest.java

```diff
@@ -0,0 +1,91 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import java.util.List;
+import java.util.Random;
+
+import static SixPMComTestSuite.utils.SixPmConstants.*;
+
+public class SixPMFeaturesTest {
+    private WebDriver driver;
+
+    @BeforeClass
+    public void setDriver() {
+        System.setProperty("webdriver.chrome.driver", "src/test/resources/chromedriver.exe");
+    }
+
+    @BeforeMethod
+    public void setChromeDriver(){
+        driver = new ChromeDriver();
+    }
+
+    @AfterMethod
+    public  void closeDriver(){
+        WebDriverWait wait = new WebDriverWait(driver,10);
+
+        WebElement itemQuantityDropDown = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(itemQuantityDropDownXpath)));
+        itemQuantityDropDown.click();
+        WebElement removeButtonFromDropDown = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(removeButtonFromDropDownXpath)));
+        removeButtonFromDropDown.click();
+
+        driver.quit();
+    }
+
+    @Test
+    public void addToFavTest(){
+        WebDriverWait wait = new WebDriverWait(driver,10);
+        Actions action = new Actions(driver);
+        Random random = new Random();
+
+        driver.manage().window().maximize();
+        driver.get(homePage);
+
+        WebElement accessoriesElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(accessoriesXPath)));
+        action.moveToElement(accessoriesElem).build().perform();
+
+        WebElement aviatorsButtonElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(aviatorsButtonXPath)));
+        aviatorsButtonElem.click();
+
+        List <WebElement> listOfAllItems = wait.until(ExpectedConditions.numberOfElementsToBe(By.xpath(listOfAllItemsXPath),100));
+
+        int countOfItems = listOfAllItems.size();
+        int randomItemIndex = random.nextInt(countOfItems);
+
+        List <WebElement> actItemNameElem = wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath(itemNameElemActual)));
+        String actualItemName = actItemNameElem.get(randomItemIndex).getText();
+        List <WebElement> actItemInfoElem = wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath(itemInfoElemActual)));
+        String actualItemInfo = actItemInfoElem.get(randomItemIndex).getText();
+        List <WebElement> actItemPriceElem = wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath(itemPriceElemActual)));
+        String actualItemPrice = actItemPriceElem.get(randomItemIndex).getText();
```

Comment:
Why to wait all of these elements, if you already waited for their count to be 100 ? Just get what you want without waiting.

Reply:
changed